test/test_viz.py

```diff
@@ -0,0 +1,48 @@
+import jax.numpy as jnp
+import pygame
+import pygame.locals
+
+from cotix._bodies import Ball
+from cotix._visualizer_tools import BallDrawer, Visualizer
+
+
+def test_viz():
+    framerate = 60
+
+    viz = Visualizer(100, 6, 6, name="Test Viz")
+
+    b = Ball.make_default()
+
+    b = b.set_position(b.position + jnp.array([3.0, 3.0]))
+
+    print(b.position)
+
+    b_drawer = BallDrawer(b, pygame.Color(255, 0, 0))
+
+    run = True
+    framecount = 0
+    inc = jnp.array([0.1, 0.1])
+
+    while run:
+        # Iterating over all the events received from
+        # pygame.event.get()
+        for event in viz.get_events():
+            # If the type of the event is quit
+            # then setting the run variable to false
+            if event.type == pygame.locals.QUIT:
+                run = False
+        # Draws the surface object to the screen.
+
+        viz.draw([b_drawer])
```

Comment:
Why do you have to draw b_drawer? Just draw all the objects passed. Like,
```viz.draw([Ball(), Robot(), ...])``` and viz draw could look something like
```
for x in objects:
    draw_body(x)
```

Reply:
Because the information in the objects `Ball(), Robot()` is not enough to draw them, e.g. the color. We can add this data as properties of the objects, but as this is only visualization related, I think it is better to create other classes called `BallDrawer(), RobotDrawer()` that draw the corresponding object according to additional data passed

Reply:
This is not exactly visualization related: color is in fact a property of a body. While I cannot think about any other ways to use it, the rule of thumb is that if something is a property of a particular type of objects, you should include this something (color in this case) as property of the class. This is just sensible: while we indeed are going to perceive a performance hit, in practice it is firstly really small, and, secondly, there is a way to avoid it completely by marking the color as static: then this property won't be passed around while doing any operations with objects, which means 0 performance degradation.

I think that making a color of a body an ```Optional[Color]``` or just ```Color``` is a good enough decision. If you don't believe me, well, alright, you can open a discussion about it in the discussions tabs of the repo. 

I am ok with doing it your way if you can give a sensible explanation why is introducing a hierarchy of drawing classes is better than using ```Color``` as a property. For now I don't see why you would do this.

Reply:
Hm, your point about showing something different when complicated situations occur does make sense. I guess I was thinking that users could use cotix to make any environment they would like. 

There is a nice abstraction in Jax that is usually used to do that, but I already messed it up a bit, so doing it your way is quite a decent solution.

To sum up: do it in any way you'd like. I stopped having a strong opinion on the topic.